crates/cli/src/codegen/mod.rs

```diff
@@ -197,8 +247,15 @@ impl Generator {
                     memory,
                 ))
             }
-            CodeGenType::Dynamic => {
-                let import_namespace = self.plugin.import_namespace()?;
+            LinkingKind::Dynamic => {
+                // All plugins by default are assumed to be linking against default
+                // or no module. For plugins linking against the V2 module we need
+                // the import_namespace will always be hardcoded to the same value.
+                let import_namespace = match self.plugin_kind {
+                    plugin::PluginKind::V2 => "javy_quickjs_provider_v2".to_string(),
+                    _ => self.plugin.import_namespace()?,
+                };
```

Comment:
So this kind of sucks... I'll admit that but I couldn't think of a better way to keep the `import_namespace` function in this crate while supporting the existing behavior/tests.

Reply:
Could you add a function like `plugin_kind.import_namespace(&plugin) -> String` to `PluginKind` with this implementation? It's moving logic around but at least that way the namespace will be in the `plugin` module as opposed to the generator.

Reply:
I like that quite a bit more.